showcase/app/templates/components/application-state.hbs

```diff
@@ -7,124 +7,494 @@
-    <SF.Item @label="Error state (ideal)">
+<section data-test-percy class="{{if this.showHighlight 'shw-component-application-state-layout-highlight'}}">
+
+  <Shw::Text::H2>Content</Shw::Text::H2>
+
+  <button type="button" class="shw-component-application-state-button-highlight" {{on "click" this.toggleHighlight}}>
+    {{if this.showHighlight "Hide" "Show"}}
+    layout highlight
+  </button>
+
+  <Shw::Flex @direction="column" @gap="2rem" as |SF|>
+    <SF.Item @label="Without actions">
+      <Hds::ApplicationState as |A|>
+        <A.Header @title="Empty state title" />
+        <A.Body
+          @text="Sorry, we couldn't find any results matching your search criteria. Please try again with different search terms or refine your filters."
+        />
+      </Hds::ApplicationState>
+    </SF.Item>
+    <SF.Item @label="With link action">
+      <Hds::ApplicationState as |A|>
+        <A.Header @title="Empty state title text" />
+        <A.Body
+          @text="Sorry, we couldn't find any results matching your search criteria. Please try again with different search terms or refine your filters."
+        />
+        <A.Footer as |F|>
+          <F.LinkStandalone @icon="arrow-left" @text="Go back" @href="/" />
+        </A.Footer>
+      </Hds::ApplicationState>
+    </SF.Item>
+  </Shw::Flex>
+
+  <Shw::Divider @level={{2}} />
+
+  <Shw::Flex @direction="column" @gap="2rem" as |SF|>
+    <SF.Item @label="With generic body content">
+      <Hds::ApplicationState as |A|>
+        <A.Header @title="Empty state title text" />
+        <A.Body>
+          <Shw::Placeholder @text="generic content" @width="320px" @height="120px" />
+        </A.Body>
+      </Hds::ApplicationState>
+    </SF.Item>
+    <SF.Item @label="With generic content + Footer with action">
+      <Hds::ApplicationState as |A|>
+        <A.Header @title="Empty state title text" />
+        <A.Body>
+          <Shw::Placeholder @text="generic content" @width="320px" @height="120px" />
+        </A.Body>
+        <A.Footer as |F|>
+          <F.LinkStandalone @icon="arrow-left" @text="Go back" @href="/" />
+        </A.Footer>
+      </Hds::ApplicationState>
+    </SF.Item>
+  </Shw::Flex>
+
+  <Shw::Divider @level={{2}} />
+
+  <Shw::Flex @direction="column" @gap="2rem" as |SF|>
+    <SF.Item @label="With icon in header">
+      <Hds::ApplicationState as |A|>
+        <A.Header @title="An error has occurred" @icon="help" />
+        <A.Body
+          @text="Sorry, we couldn't find any results matching your search criteria. Please try again with different search terms or refine your filters."
+        />
+        <A.Footer as |F|>
+          <F.LinkStandalone @icon="arrow-left" @text="Go back" @href="/" />
+        </A.Footer>
+      </Hds::ApplicationState>
+    </SF.Item>
+    <SF.Item @label="With error code in header">
       <Hds::ApplicationState as |A|>
         <A.Header @title="An error has occurred" @icon="alert-circle" @errorCode="404" />
```

Comment:
So we can test the use case when there's an error code but no icon (is a different CSS rule)
```suggestion
        <A.Header @title="An error has occurred" @errorCode="404" />
```